generate_figures/linear_overfit_table.py

```diff
@@ -1,88 +1,23 @@
-    else: 
-        df= pd.concat([df,pd.DataFrame(array_x, columns=cols_df)], axis =0)
-    df = df.reset_index(drop=True)
-
-# Preparing the results for the table
-
-#Finding the ones that converged
-df_converge = df[df["test_u_rel_error"]< 0.01] # Convergence criteria : test_u_rel_error < 0.01
-# Calculating the succes rates
-data_points= sorted(df_converge['Number of data points'].unique().tolist())
-data_points = [int(x) for x in data_points]
-
-success_rate = []
-for i in data_points:
-    success = 100*len(df_converge[df_converge["Number of data points"] == i])/len(df[df["Number of data points"] == i])
-    success_rate.append(success)
-# Creating the result data frame    
-df_results = df_converge.groupby("Number of data points").median()
-df_results['test_u_rel_error'] = df_results['test_u_rel_error']*100
-df_results = df_results.drop(['retcode'], axis=1)
-df_results['success'] = success_rate
-df_results['Number of data points'] = data_points
-df_results = df_results[['Number of data points','success', 'train_loss', 'test_loss', 'test_u_rel_error']]
-df_results = df_results.set_index(['Number of data points'])
-# Creating the latex file
-
-
-def latex_table(df):
-    df = df.rename(
-            columns={
-                "success":r"\shortstack{Success \\(\%)}",
-                "train_loss": r"\shortstack{Train MSE \\ ($\varepsilon$)}",
-                "test_loss": r"\shortstack{Test MSE \\ ($\varepsilon$)}",
-                "test_u_rel_error": r"\shortstack{Policy Error\\ ($\epsilon_{\mathrm{rel}}$)}",
-            }
-        )
-
-    latex_str = df.to_latex(
-        multicolumn=True,
-        multirow=True,
-        formatters=[
-            "{:0.0f}\%".format,
-            "{:.1e}".format,
-            "{:.1e}".format,
-            "{:.2f}\%".format
-        ],
-        index_names =  "number of ", 
-        longtable=False,
-        sparsify=True,
-        escape=False,
-    )
-    latex_list = latex_str.splitlines()
-    #latex_list.insert(6, '\midrule')
-    latex_new = '\n'.join(latex_list)
-    return latex_new
+project = "highdimensionaleconlab/symmetry_dynamic_programming"
+
+df_results = pd.DataFrame()
+df = get_results_by_tag(api, project, "deep_sets_overfit", get_config=True)
+assert df.id.nunique() == 400
+df = df.rename(columns={"model.train_subsample_trajectories": "Number of data points"})
+df_results = pd.DataFrame()
+df_results["success"] = df[df["retcode"] == 0].groupby("Number of data points").count()["id"]
+df_results[["train_loss", "test_loss"]] = (
+    df[df["retcode"] == 0].groupby("Number of data points")[["train_loss", "test_loss"]].median()
+)
+df_results["test_u_rel_error"] = (
+    df[df["retcode"] == 0].groupby("Number of data points")["test_u_rel_error"].median() * 100
+)
 
 with open(output_dir + "/linear_overfit_table.tex", "w") as file:
-    file.write(latex_table(df_results))
\ No newline at end of file
+    file.write(df_to_latex(df_results))
```

Comment:
My old table was wrong. I think this new one is correct, I tested it by manually going onto W&B and checking the successes and then randomly grabbed test loss for trajectory = 4 and took the median. Both these successes and test loss aligned with my new table and not the old. I am pretty positive this one is correct but wanted to highlight it since it changed.

Reply:
@Mekahou  You may want to note this and add an issue to check.  Resolvingt the conversation